src/core/src/index/revindex/mod.rs

```diff
@@ -425,20 +425,20 @@ impl Datasets {
         }
     }
 
-    fn from_slice(slice: &[u8]) -> Option<Self> {
+    pub fn from_slice(slice: &[u8]) -> Result<Self> {
         use byteorder::ReadBytesExt;
 
         if slice.len() == 8 {
             // Unique
-            Some(Self::Unique(
+            Ok(Self::Unique(
                 (&slice[..]).read_u32::<LittleEndian>().unwrap(),
             ))
         } else if slice.len() == 1 {
             // Empty
-            Some(Self::Empty)
+            Ok(Self::Empty)
         } else {
             // Many
-            Some(Self::Many(RoaringBitmap::deserialize_from(slice).unwrap()))
+            Ok(Self::Many(RoaringBitmap::deserialize_from(slice).unwrap()))
```

Comment:
Same as above, might need new error and not strictly needed now

```suggestion
            Ok(Self::Many(RoaringBitmap::deserialize_from(slice)?))
```

Reply:
same!